src/operation.js

```diff
@@ -358,14 +368,16 @@ export class Operation {
         if (!isString(opts.amount)) {
             throw new TypeError('amount argument must be of type String');
         }
-        attributes.amount = Hyper.fromString(String(opts.amount));
+        let amount = new BigNumber(opts.amount).mul(ONE);
+        attributes.amount = Hyper.fromString(amount.toString());
         if (isUndefined(opts.price)) {
             throw new TypeError('price argument is required');
         }
-        let approx = best_r(opts.price);
+        let price = new BigNumber(opts.price);
+        let approx = price.toFraction(MAX_DENOMINATOR);
         attributes.price = new xdr.Price({
-            n: approx[0],
-            d: approx[1]
+            n: parseInt(approx[0]),
+            d: parseInt(approx[1])
```

Comment:
This block above is copy-pasta from the line 320-ish area.  Please extract into a helper method.